src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java

```diff
@@ -279,24 +256,29 @@ public void unstash(String name, FilePath workspace, Launcher launcher, EnvVars
                     String.format("No such saved stash ‘%s’ found at %s/%s", name, provider.getContainer(), blobPath));
         }
         URL url = provider.toExternalURL(blob, HttpMethod.GET);
-        workspace.act(new Unstash(url));
+        workspace.act(new Unstash(url, listener));
         listener.getLogger().printf("Unstashed file(s) from %s%n", provider.toURI(provider.getContainer(), blobPath));
     }
 
     private static final class Unstash extends MasterToSlaveFileCallable<Void> {
         private static final long serialVersionUID = 1L;
         private final URL url;
+        private final TaskListener listener;
+        private final RobustHTTPClient client = JCloudsArtifactManager.client;
 
-        Unstash(URL url) throws IOException {
+        Unstash(URL url, TaskListener listener) throws IOException {
             this.url = url;
+            this.listener = listener;
         }
 
         @Override
         public Void invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
-            try (InputStream is = url.openStream()) {
-                new FilePath(f).untarFrom(is, FilePath.TarCompression.GZIP);
-                // Note that this API currently offers no count of files in the tarball we could report.
-            }
+            client.connect("download", "download " + url.toString().replaceFirst("[?].+$", "?…") + " into " + f, c -> c.execute(new HttpGet(url.toString())), response -> {
```

Comment:
`…` vs `...` ?
should replace passwords in `http://user:password@...`
wouldn't be easier to parse into `Url` and pick the sections needed rather than regex?

Reply:
Oops, forgot to use `sanitize` here!